cmd/permissions-checker/permissionschecks.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os/user"
+	"regexp"
+	"strings"
+	"syscall"
+)
+
+// FindRecursive iterate in a directory (path) showing permissions in a recursive way
+/* path  - where to look for files and directories
+ * level - used to print the output in a hierarchical way
+ */
+func FindRecursive(path, level string) {
+	files, err := ioutil.ReadDir(path)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// In verbose mode we are printing the output in a hierarchical way
+	if Verbose {
+		level = level + "  "
+	}
+
+	for _, f := range files {
+		name := f.Name()
+		relativePath := strings.TrimPrefix(strings.Join([]string{path, name}, "/"), Directory)
+		fullPath := strings.Join([]string{path, name}, "/")
+
+		match, _ := regexp.MatchString(Exclude, relativePath)
```

Comment:
This could be written as:
```
if match, _ := regexp.MatchString(Exclude, relativePath); match {
    if Verbose {
        fmt.Printf("\x1b[33m%sExcluding %s\n\x1b[0m", level, name)
    }
    continue
}
// ... current else block
mode := f.Mode()
currentOwner, _ := user.LookupId(fmt.Sprint(f.Sys().(*syscall.Stat_t).Uid))
```